src/fsql/column_parser.py

```diff
@@ -84,13 +84,25 @@ def process_single_partition(partition_desc: str):
 
 
 class AutoParser(ColumnParser):
-    def __init__(self, partition_grammars: Optional[list[PartitionGrammar]] = None):
+    def __init__(self, partition_grammars: Optional[list[PartitionGrammar]] = None, parse_filenames: bool = False):
```

Comment:
I'd vote for keeping this simpler via only `parse_filenames_as: Optional[str] = None`, instead of a bool param and an internal str with setter/getter. The `parses_filenames` would change to `return self.parse_filenames_as is not None`, the return value in `__call__` change to `self.parse_filename_as`.

Reply:
I was looking exactly for sth like that, I just was not able to come up with a reasonable name, thanks!